osquery/dispatcher/windows/query_profiler.cpp

```diff
@@ -0,0 +1,37 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <chrono>
+
+#include <boost/format.hpp>
+
+#include <osquery/dispatcher/query_profiler.h>
+#include <osquery/killswitch.h>
+#include <osquery/numeric_monitoring.h>
+
+namespace osquery {
+void launchQueryWithProfiling(const std::string& name,
+                              std::function<Status()> launchQuery) {
+  const auto start_time_point = std::chrono::steady_clock::now();
```

Comment:
Why do you calculate current time time twice if `Killswitch::get().isWindowsProfilingEnabled()` is false? Would you like to do something like posix implementation here?